packages/block-editor/src/components/copy-handler/index.js

```diff
@@ -122,7 +122,10 @@ export function useClipboardHandler() {
 				}
 				notifyCopy( event.type, selectedBlockClientIds );
 				const blocks = getBlocksByClientId( selectedBlockClientIds );
-				const serialized = serialize( blocks );
+				const serialized = serialize( blocks, {
+					__experimentalExcludeNonCopyableAttributes:
+						event.type === 'copy',
```

Comment:
this one feels overly specific when we're introducing a general way we could accomplish this too.

```ts
const serializeOptiosn = event.type === 'copy'
	? { __experimentalIncludeAttributes: [ 'copy' ]
	: undefined;

const serialized = serialize( blocks, serializeOptions );
```

---

Also I'm really lost on the double negative. A positive phrasing could make that easier to understand.

Reply:
(this would give us the ability to denote multiple sets of attributes for inclusion _or_ sets for explicit exclusion)

```ts
{
	__experimentalExcludeAttributes: [ 'style' ]
}
```

Reply:
The variable name definitely leaves a _lot_ to be desired, suggestions are very much appreciated 😄

Making it extensible to other sets of attributes is great! It gets a little busy if we want to include the ability to filter out by _any_ attribute property (rather than limiting it to a list of `supports` keys for example):

```
const serializationOptions =  {
    __experimentalIncludeAttributes: {
            __experimentalSupports: { copy: false },
    }
}
const serialized = serialize(
    blocks,
    event.type === 'copy' ? serializationOptions : undefined
);
```

Not sure much can be done about that 🤔 I think the benefit to extensibility and the clearer naming is a big win.

Reply:
just a note about my example: I wasn't proposing nesting `supports` inside of `includeAttributes`. using your object notation instead of an array it could be this too, which is closer to what I was trying to imply. the include/exclude arrays have some advantages, but they aren't directly necessary here either

```ts
const options = {
	__experimentalAttributeSupports: {
		copy: false,
	}
}
```

Reply:
This is tricky. Nesting `supports` inside of `includeAttributes` is very verbose but reads clearly to me. I'm a little concerned with this:

```
// Is it clear from this that attributes that do not support `copy` will be omitted?
cloneBlock(
    block,
    __experimentalOptions: {
        __experimentalSupports: { copy: false }
    }
}

// Would it be easy to confuse with this -- here attributes that DO support `copy` would be omitted:
cloneBlock(
    block,
    __experimentalOptions: {
        __experimentalSupports: { copy: true }
    }
}
```

Sorry if I'm misunderstanding, the double negatives are confusing. 

```
// This is more 'streamlined' and reads more clearly to me, but only supports excluding attributes by `supports` keys. Eg you can't exclude attributes by their role for example
cloneBlock(
    block,
    __experimentalOptions: {
        __experimentalExcludeAttributes: { copy: false }
    }
}
```